### PR TITLE
feat: Create & Mint badge flow - Profile and Badge Preview draft view

### PR DESCRIPTION
**Pull request by FedeMadoery**
# Description

This PR has a bunch of small changes here and there, to make the forms for Badge Type creation and Bage Mint as much usable as possible. It is possible that some forms o fields are not 100% accurate with the designs, the idea of this PR is to make it usable and kind of "stable". The UI polish will be done later on. 

In addition, I added two more pages, the` /profile` one and a badge view `/ {id} / {ownerId}`

## What's new?

- [x] Fix some issues
- [x] Improve labels
- [x] Improve create badge forms
- [x] Improve form fields
- [x] Profile list of badges (drag view)
- [x] Badge preview (draft view)

## How should this PR be Tested?

Please describe the tests that you ran to verify your changes.

- [ ] Test A
- [ ] Test B

# Screenshots (if relevant)

### CODE REVIEW

File: pages/badge/[typeId]/[ownerAddress].tsx [hunk 1]
```diff
@@ -1,18 +1,71 @@
...
-      </Typography>
+      <Box display="flex" flex={1} gap={4} justifyContent="space-evenly" my={4}>
+        <Box display="flex">
+          <BadgeTypeMetadata metadata={badgeType?.metadataURL} />
+        </Box>
+        <Stack gap={3} justifyContent="space-between">
+          <Stack gap={2}>
+            <Typography
+              sx={{
+                color: '#22DBBD',
```
Comment (ramabit) [inline pages/badge/[typeId]/[ownerAddress].tsx:35]: please add color in theme and take it from there

File: pages/badge/[typeId]/[ownerAddress].tsx [hunk 2]
```diff
@@ -1,18 +1,71 @@
...
+                fontSize: '25px !important',
+                fontWeight: 'bold',
+                textShadow: '0px 0px 7px rgba(51, 255, 204, 0.6)',
+              }}
+            >
+              Title Github Passport
+            </Typography>
+            <Typography
+              sx={{ color: '#22DBBD', fontSize: 10, fontWeight: 'bold' }}
```
Comment (ramabit) [inline pages/badge/[typeId]/[ownerAddress].tsx:44]: please add color in theme and take it from there

File: pages/badge/[typeId]/[ownerAddress].tsx [hunk 3]
```diff
@@ -1,18 +1,71 @@
...
+              variant="caption"
+            >
+              BadgeID: {typeId}
+            </Typography>
+          </Stack>
+          <Divider color={colors.white} />
+          <Stack gap={2}>
+            <Typography variant="body2">Issued by: TheBadge</Typography>
+            <Typography variant="body2">
+              Minting your GitHub Passport is the first step to getting started with
```
Comment (ramabit) [inline pages/badge/[typeId]/[ownerAddress].tsx:54]: please take text from translations

File: pages/badge/[typeId]/[ownerAddress].tsx [hunk 4]
```diff
@@ -1,18 +1,71 @@
...
+            <Typography
+              sx={{ color: '#22DBBD', fontSize: 10, fontWeight: 'bold' }}
+              variant="caption"
+            >
+              BadgeID: {typeId}
+            </Typography>
+          </Stack>
+          <Divider color={colors.white} />
+          <Stack gap={2}>
+            <Typography variant="body2">Issued by: TheBadge</Typography>
```
Comment (ramabit) [inline pages/badge/[typeId]/[ownerAddress].tsx:52]: please take text from translations

File: pages/badge/[typeId]/[ownerAddress].tsx [hunk 5]
```diff
@@ -1,18 +1,71 @@
...
+            <Typography variant="body2">
+              Minting your GitHub Passport is the first step to getting started with
+              Proof-of-Talent. Claim the badge by connecting your GitHub account to Kleoverse and
+              prove it is you who’s earning credentials. GitHub is where over 83 million developers
+              shape the future of software, together. Show off your programming skills by building
+              software and contributing to open-source projects. Claimable once.
+            </Typography>
+          </Stack>
+          <Divider color={colors.white} />
+          <Typography>{badgeType?.badgesMintedAmount} # of claims</Typography>
```
Comment (ramabit) [inline pages/badge/[typeId]/[ownerAddress].tsx:62]: please take text from translations

File: pages/badge/[typeId]/[ownerAddress].tsx [hunk 6]
```diff
@@ -1,18 +1,71 @@
...
+          <Box display="flex" justifyContent="space-between">
+            <Typography>Check who else claimed it SEE ALL</Typography>
```
Comment (ramabit) [inline pages/badge/[typeId]/[ownerAddress].tsx:64]: please take text from translations

File: pages/profile/[address].tsx [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+  const userWithBadges = gql.useUserBadgesById({ ownerAddress: addressOnUrl })
+  const badges = userWithBadges.data?.user?.badges || []
+
+  return (
+    <>
+      <Stack sx={{ mb: 6, gap: 4, alignItems: 'center' }}>
+        <Typography textAlign="center" variant="subtitle2">
+          {address === addressOnUrl
+            ? 'This is your profile'
+            : `This is the profile of ${addressOnUrl}`}
```
Comment (ramabit) [inline pages/profile/[address].tsx:36]: please take texts from translations

File: pages/profile/[address].tsx [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+                display: 'flex',
+                flexDirection: 'column',
+                gap: 1,
+                border: `1px solid ${colors.green}`,
+                borderRadius: 2,
+                p: 2,
+              }}
+            >
+              <LinkWithTranslation pathname={`/badge/${bt.badgeType.id}/${addressOnUrl}`}>
+                Minted badge {bt.id}
```
Comment (ramabit) [inline pages/profile/[address].tsx:61]: please take text from translations

File: pages/profile/[address].tsx [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+              </LinkWithTranslation>
+
+              <Typography>Status: {bt.status}</Typography>
+              <Typography>
+                Review DueDate: {dayjs.unix(bt.reviewDueDate).format('dddd, MMMM D, YYYY h:mm A')}
```
Comment (ramabit) [inline pages/profile/[address].tsx:66]: please take text from translations

File: pages/profile/[address].tsx [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+              </Typography>
+              <Typography>BadgeType Id: {bt.badgeType.id}</Typography>
```
Comment (ramabit) [inline pages/profile/[address].tsx:68]: please take text from translations

File: pages/profile/[address].tsx [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+              <Typography>BadgeType Minted Amount: {bt.badgeType.badgesMintedAmount}</Typography>
```
Comment (ramabit) [inline pages/profile/[address].tsx:69]: please take text from translations

File: pages/profile/index.tsx [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+  if (address) {
+    router.replace(`/profile/${address}`)
+    return null
+  }
+
+  return (
+    <>
+      <Stack sx={{ mb: 6, gap: 4, alignItems: 'center' }}>
+        <Typography textAlign="center" variant="subtitle2">
+          Please connect your wallet
```
Comment (ramabit) [inline pages/profile/index.tsx:22]: please take text from translations

File: src/components/form/ExpirationField.tsx [hunk 13]
```diff
@@ -94,17 +95,16 @@ export default function ExpirationField() {
...
-      statusText={error?.errorMessage}
+      statusText={error ? error?.errorMessage : ' '}
     />
   )
 }
 
-function validTo(unit: 'day' | 'month' | 'year', days?: z.infer<typeof ExpirationTypeSchema>) {
+function validTo(days?: z.infer<typeof ExpirationTypeSchema>) {
   if (!days || days === 0) return 'End of time.'
-  return dayjs().add(days, unit).format('DD/MM/YYYY')
```
Comment (ramabit) [inline src/components/form/ExpirationField.tsx:107]: I think we should have a general utils for date formatting

File: src/components/form/ImageInput.tsx [hunk 14]
```diff
@@ -17,11 +30,32 @@ const Wrapper = styled(Box)(({ theme }) => ({
...
+})<
+  AvatarProps & {
+    isDragging?: boolean
+  }
+>(({ isDragging, theme }) => ({
+  width: 200,
+  height: 200,
+  borderRadius: theme.spacing(2),
+  borderWidth: 1,
+  backgroundColor: 'transparent',
```
Comment (ramabit) [inline src/components/form/ImageInput.tsx:47]: take transparent color from theme

File: src/components/form/ImageInput.tsx [hunk 15]
```diff
@@ -17,11 +30,32 @@ const Wrapper = styled(Box)(({ theme }) => ({
...
+  borderColor: isDragging ? colors.green : 'white',
```
Comment (ramabit) [inline src/components/form/ImageInput.tsx:48]: please use colors.white

File: src/components/form/ImageInput.tsx [hunk 16]
```diff
@@ -59,38 +90,65 @@ export default function ImageInput() {
...
+                      isDragging={isDragging}
                       {...dragProps}
+                      onClick={onImageUpload}
+                      sx={{ cursor: 'pointer' }}
                     >
-                      Click or Drop here
-                    </Button>
+                      <Stack sx={{ flex: 1, flexWrap: 'wrap', alignContent: 'center' }}>
+                        <Typography color="text.primary" variant="subtitle2">
+                          Click here or drop
```
Comment (ramabit) [inline src/components/form/ImageInput.tsx:107]: please take text from translations

File: src/components/form/SeveritySelector.tsx [hunk 17]
```diff
@@ -19,6 +20,40 @@ const Wrapper = styled(Box)(({ theme }) => ({
   gridColumn: 'span 1 / span 2',
 }))
 
+const CustomSlider = styled(Slider)({
+  height: '5px',
+  '& .MuiSlider-thumb': {
+    [`&:hover, &.Mui-focusVisible`]: {
+      //  boxShadow: '0px 0px 0px 8px var(--box-shadow)',
```
Comment (ramabit) [inline src/components/form/SeveritySelector.tsx:27]: why is this commented?

File: src/components/form/SeveritySelector.tsx [hunk 18]
```diff
@@ -19,6 +20,40 @@ const Wrapper = styled(Box)(({ theme }) => ({
...
+    },
+    [`&.Mui-active`]: {
+      //  boxShadow: '0px 0px 0px 14px var(--box-shadow)',
```
Comment (ramabit) [inline src/components/form/SeveritySelector.tsx:30]: why is this commented?

File: src/components/form/SeveritySelector.tsx [hunk 19]
```diff
@@ -19,6 +20,40 @@ const Wrapper = styled(Box)(({ theme }) => ({
...
+    },
+  },
+  '& .MuiSlider-track': {
+    [`&:hover, &.Mui-focusVisible`]: {
+      //  boxShadow: '0px 0px 0px 8px var(--box-shadow)',
```
Comment (ramabit) [inline src/components/form/SeveritySelector.tsx:35]: why is this commented?

File: src/components/form/SeveritySelector.tsx [hunk 20]
```diff
@@ -19,6 +20,40 @@ const Wrapper = styled(Box)(({ theme }) => ({
...
+})
+
+const marks = [
+  {
+    value: 1,
+    label: Severity_Keys[0],
+  },
+  //{
+  //  value: 3,
+  //  label: Severity_Keys[1],
```
Comment (ramabit) [inline src/components/form/SeveritySelector.tsx:49]: why is this code commented?

File: src/components/form/helpers/customSchemas.ts [hunk 21]
```diff
@@ -79,6 +79,16 @@ export const LongTextSchema = createUniqueFieldSchema(
     .min(25, { message: 'Text field most have at least 25 characters.' }),
   'LongTextSchema',
 )
+export const DescriptionTextSchema = createUniqueFieldSchema(
+  z
+    .string({
+      required_error: 'Is required',
+      invalid_type_error: 'Must be an string',
+    })
+    .min(25, { message: 'Text field most have at least 25 characters.' }),
```
Comment (ramabit) [inline src/components/form/helpers/customSchemas.ts:88]: please take text from translations

File: src/components/form/helpers/customSchemas.ts [hunk 22]
```diff
@@ -123,6 +133,16 @@ export const ExpirationTypeSchema = createUniqueFieldSchema(
   'ExpirationTypeSchema',
 )
 
+export const ChallengePeriodTypeSchema = createUniqueFieldSchema(
+  z
+    .number({
+      required_error: 'Is required',
+      invalid_type_error: 'Must enter an amount of days',
+    })
+    .min(2, 'The challenge time must be greater than 1 day'),
```
Comment (ramabit) [inline src/components/form/helpers/customSchemas.ts:142]: please take text from translations

File: src/components/form/helpers/customSchemas.ts [hunk 23]
```diff
@@ -139,10 +159,10 @@ export const KlerosFormFieldSchema = z.object({
   label: z
     .string()
     .describe('Name // Field name that the user will be when they create the badge.'),
-  description: z
-    .string()
-    .describe('Description // Field description that explains what the field data is.'),
   type: KlerosFieldTypeSchema.describe('Field Type // Type of the required data.'),
+  description: LongTextSchema.describe(
+    'Description // Field description that explains what the field data is.',
```
Comment (ramabit) [inline src/components/form/helpers/customSchemas.ts:164]: please take text from translations

File: src/components/form/helpers/customSchemas.ts [hunk 24]
```diff
@@ -139,10 +159,10 @@ export const KlerosFormFieldSchema = z.object({
   label: z
     .string()
     .describe('Name // Field name that the user will be when they create the badge.'),
```
Comment (ramabit) [inline src/components/form/helpers/customSchemas.ts:161]: please take text from translations

File: src/components/form/helpers/customSchemas.ts [hunk 25]
```diff
@@ -139,10 +159,10 @@ export const KlerosFormFieldSchema = z.object({
...
-  description: z
-    .string()
-    .describe('Description // Field description that explains what the field data is.'),
   type: KlerosFieldTypeSchema.describe('Field Type // Type of the required data.'),
```
Comment (ramabit) [inline src/components/form/helpers/customSchemas.ts:162]: please take text from translations

File: src/components/form/klerosDynamicFormField/FormFieldCreator.tsx [hunk 26]
```diff
@@ -111,9 +111,24 @@ export default function KlerosDynamicFieldsCreator() {
       </Box>
       {error && <FormStatus status={TextFieldStatus.error}>{error.errorMessage}</FormStatus>}
       <Box>
-        <DndProvider backend={HTML5Backend}>
-          {field?.value?.map((field, index) => renderFieldItem(field, index))}
-        </DndProvider>
+        <Box sx={{ display: 'flex', justifyContent: 'space-evenly', mb: 4, ml: 5, mr: 5 }}>
+          <Typography sx={{ flex: 1 }} variant="subtitle1">
+            Name
```
Comment (ramabit) [inline src/components/form/klerosDynamicFormField/FormFieldCreator.tsx:116]: please take text from translations

File: src/components/form/klerosDynamicFormField/FormFieldCreator.tsx [hunk 27]
```diff
@@ -111,9 +111,24 @@ export default function KlerosDynamicFieldsCreator() {
...
+          </Typography>
+
+          <Typography sx={{ flex: 2 }} variant="subtitle1">
+            Description
```
Comment (ramabit) [inline src/components/form/klerosDynamicFormField/FormFieldCreator.tsx:120]: please take text from translations

File: src/components/form/klerosDynamicFormField/FormFieldCreator.tsx [hunk 28]
```diff
@@ -111,9 +111,24 @@ export default function KlerosDynamicFieldsCreator() {
...
+          </Typography>
+
+          <Typography sx={{ flex: 1 }} variant="subtitle1">
+            Type
```
Comment (ramabit) [inline src/components/form/klerosDynamicFormField/FormFieldCreator.tsx:124]: please take text from translations

File: src/pagePartials/badge/mint/MintSteps.tsx [hunk 29]
```diff
@@ -23,42 +31,86 @@ type MintStepsProps<SchemaType extends z.ZodEffects<any, any, any> | AnyZodObjec
...
+          <Typography>Mint cost: {costs.mintCost}.</Typography>
+          <Box alignItems={'center'} display="flex">
+            Deposit for Kleros: {costs.klerosCost}.
+            <Tooltip title={'This will be returned if the evidence is valid'}>
+              <InfoOutlinedIcon />
+            </Tooltip>
+          </Box>
+          <Typography>Total (Native token) need: {costs.totalMintCost}.</Typography>
+        </Stack>
+        <Button onClick={convertPreviewToImage}>Convert Preview to Image</Button>
```
Comment (ramabit) [inline src/pagePartials/badge/mint/MintSteps.tsx:113]: please take text from translations

File: src/pagePartials/badge/mint/MintSteps.tsx [hunk 30]
```diff
@@ -23,42 +31,86 @@ type MintStepsProps<SchemaType extends z.ZodEffects<any, any, any> | AnyZodObjec
...
+        </Box>
+        <Stack>
+          <Typography>Mint cost: {costs.mintCost}.</Typography>
+          <Box alignItems={'center'} display="flex">
+            Deposit for Kleros: {costs.klerosCost}.
+            <Tooltip title={'This will be returned if the evidence is valid'}>
+              <InfoOutlinedIcon />
+            </Tooltip>
+          </Box>
+          <Typography>Total (Native token) need: {costs.totalMintCost}.</Typography>
```
Comment (ramabit) [inline src/pagePartials/badge/mint/MintSteps.tsx:111]: please take text from translations

File: src/pagePartials/badge/mint/MintSteps.tsx [hunk 31]
```diff
@@ -23,42 +31,86 @@ type MintStepsProps<SchemaType extends z.ZodEffects<any, any, any> | AnyZodObjec
...
+            size="medium"
+            textContrast="light-withTextBackground"
+            title={badgeMetadata.name}
+          />
+        </Box>
+        <Stack>
+          <Typography>Mint cost: {costs.mintCost}.</Typography>
+          <Box alignItems={'center'} display="flex">
+            Deposit for Kleros: {costs.klerosCost}.
+            <Tooltip title={'This will be returned if the evidence is valid'}>
```
Comment (ramabit) [inline src/pagePartials/badge/mint/MintSteps.tsx:107]: please take text from translations

File: src/pagePartials/badge/type/CreateSteps.tsx [hunk 32]
```diff
@@ -0,0 +1,131 @@
...
+} from '@/src/components/form/helpers/customSchemas'
+
+const MintSchemaStep1 = z.object({
+  help: AgreementSchema.describe(`How it works // ??`),
+})
+
+const MintSchemaStep2 = z.object({
+  name: z.string().describe('Name // ??'),
+  description: DescriptionTextSchema.describe(
+    'Description // This description will be showed on the Badge itself, you can use some helpers to inject user information on it.',
```
Comment (ramabit) [inline src/pagePartials/badge/type/CreateSteps.tsx:29]: please take all texts in this file from translations

**Review by ramabit — CHANGES_REQUESTED** [review]

File: src/utils/kleros/generateKlerosListMetaEvidence.ts [hunk 33]
```diff
@@ -40,13 +40,13 @@ export function generateKlerosListMetaEvidence(
...
   // TODO
   // check max items indexed = 3
 
   const itemNamePlural = `${badgeName}s`
 
-  const registration: klerosListStructure = {
+  const registration: KlerosListStructure = {
     name: badgeTypeName,
     title: `${badgeTypeName} evidences.`,
     description: `Add the evidence of the badge ${badgeName} to the list of evidences of ${badgeTypeName}.`,
```
Comment (ramabit) [inline src/utils/kleros/generateKlerosListMetaEvidence.ts:52]: please take all texts in this file from translations

**Review by ramabit — CHANGES_REQUESTED** [review]

File: src/components/form/SeveritySelector.tsx (hunk 17 above)
Comment (FedeMadoery) [reply src/components/form/SeveritySelector.tsx:27]: 🧹

File: src/components/form/SeveritySelector.tsx (hunk 19 above)
Comment (FedeMadoery) [reply src/components/form/SeveritySelector.tsx:35]: 🧹

**Comment by FedeMadoery** [comment]
I move the most important stuff to the translation files and created tasks on the backlog for the other ones 👍

**Pull request merged**